src/functionalTest/java/uk/gov/hmcts/reform/pbis/utils/DeadLetterQueueHelper.java

```diff
@@ -0,0 +1,91 @@
+package uk.gov.hmcts.reform.pbis.utils;
+
+import com.microsoft.azure.servicebus.ClientFactory;
+import com.microsoft.azure.servicebus.IMessage;
+import com.microsoft.azure.servicebus.IMessageReceiver;
+import com.microsoft.azure.servicebus.ReceiveMode;
+import com.microsoft.azure.servicebus.primitives.ConnectionStringBuilder;
+import com.microsoft.azure.servicebus.primitives.ServiceBusException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+
+public class DeadLetterQueueHelper implements AutoCloseable {
+
+    private static final Logger logger = LoggerFactory.getLogger(DeadLetterQueueHelper.class);
+    private final IMessageReceiver messageReceiver;
+    private final Duration maxReceiveWaitTime;
+
+    public DeadLetterQueueHelper(
+        String namespaceConnectionString,
+        String topicName,
+        String subscriptionName,
+        Duration maxReceiveWaitTime) throws ServiceBusException, InterruptedException {
+
+        this.messageReceiver = buildMessageReceiver(
+            namespaceConnectionString,
+            topicName,
+            subscriptionName
+        );
+
+        this.maxReceiveWaitTime = maxReceiveWaitTime;
+    }
```

Comment:
Configuration parameters are used only to create `IMessageReceiver`. Why not to require it only necessary parameters and let others handle the configuration? Here is the factory method, which also could be moved to DI config.

```java
    public DeadLetterQueueHelper(IMessageReceiver messageReceiver, Duration maxReceiveWaitTime) {
        this.messageReceiver = messageReceiver;
        this.maxReceiveWaitTime = maxReceiveWaitTime;
    }

    public static DeadLetterQueueHelper create(
        String namespaceConnectionString,
        String topicName,
        String subscriptionName,
        Duration maxReceiveWaitTime) throws ServiceBusException, InterruptedException {

        String deadLetterQueuePath = String.format(
            "%s/subscriptions/%s/$DeadLetterQueue",
            topicName,
            subscriptionName
        );

        ConnectionStringBuilder connectionStringBuilder = new ConnectionStringBuilder(
            namespaceConnectionString,
            deadLetterQueuePath
        );

        IMessageReceiver messageReceiver = ClientFactory.createMessageReceiverFromConnectionStringBuilder(
            connectionStringBuilder,
            ReceiveMode.PEEKLOCK
        );

        return new DeadLetterQueueHelper(messageReceiver, maxReceiveWaitTime);
    }
```

Reply:
This is a test tool. It doesn't live in a spring application and isn't managed by a DI framework. Regardless of the approach, the tests need to provide the same parameters. I don't see any benefit.